extensions-contrib/kafka-emitter/src/main/java/org/apache/druid/emitter/kafka/KafkaEmitter.java

```diff
@@ -170,30 +189,69 @@ public void emit(final Event event)
       Map<String, Object> result = resultBuilder.build();
 
       try {
-        String resultJson = jsonMapper.writeValueAsString(result);
-        ObjectContainer<String> objectContainer = new ObjectContainer<>(
-            resultJson,
-            StringUtils.toUtf8(resultJson).length
+        byte[] resultBytes = jsonMapper.writeValueAsBytes(result);
+        ObjectContainer<byte[]> objectContainer = new ObjectContainer<>(
+            resultBytes,
+            resultBytes.length
         );
         if (event instanceof ServiceMetricEvent) {
-          if (!metricQueue.offer(objectContainer)) {
+          if (config.getMetricTopic() == null || !metricQueue.offer(objectContainer)) {
             metricLost.incrementAndGet();
           }
         } else if (event instanceof AlertEvent) {
-          if (!alertQueue.offer(objectContainer)) {
+          if (config.getAlertTopic() == null || !alertQueue.offer(objectContainer)) {
             alertLost.incrementAndGet();
           }
         } else if (event instanceof RequestLogEvent) {
           if (config.getRequestTopic() == null || !requestQueue.offer(objectContainer)) {
             requestLost.incrementAndGet();
           }
+        } else if (event instanceof SegmentMetadataEvent) {
+          if (config.getSegmentMetadataTopic() != null && config.getSegmentMetadataTopicFormat().equals(KafkaEmitterConfig.PROTOBUF)) {
```

Comment:
this code assumes that if something is not protobuf it must be json, which is a little brittle. Maybe it would be cleaner to make PROTOBUF/JSON part of an enum and have a switch case to ensure that we fail if we encounter a format we do now know, rather than corrupting data.